### PR TITLE
Mccalluc/delete tmp dirs on purge

### LINKED ISSUE #61: Remove directories on hosts when containers are purged?
With the introduction of `DockerContainerSpec.extra_directories`, directories get created on the host system upon a `run()`. 

Containers may fill these up withs lots of data and they should _probably_ be removed during a `purge_by_label()`.

### LINKED ISSUE #67: Kill containers after n minutes

### LINKED ISSUE #68: Kill containers after n minutes -> of inactivity

### PR DESCRIPTION
**Pull request by mccalluc**
Towards https://github.com/refinery-platform/refinery-platform/issues/2160 (Will need to pin the version here and update the dependency there.)

Fix #68, Fix #67 (I think)

fix #61

### CODE REVIEW

**Comment by mccalluc** [comment]
not the behavior I saw locally, but even apart from that, I'm doing things in the wrong order: should not delete directory while the container that depends on it is still running.

**Comment by mccalluc** [comment]
@scottx611x : Tests pass locally. Started off just cleaning up the temp dir, and then continued into making the temp dir configurable. If this passes, I'll tag it, and work on a separate PR in refinery proper.

**Comment by mccalluc** [comment]
Failure is related to tag mismatch, but I don't really understand it. I would merge and then release a new tag if necessary.

File: tests/test_docker_utils.py [hunk 1]
```diff
@@ -195,6 +193,7 @@ def test_purge(self):
         ))
         self.assertEqual(1, self.count_containers())
         self.assert_loads_eventually(url, 'Welcome to nginx!')
+        self.assertGreater(self.ls_tmp(), ls_tmp_orig)
```
Comment (scottx611x) [inline tests/test_docker_utils.py]: ls_tmp_orig -> self.initial_tmp

Comment (mccalluc) [reply tests/test_docker_utils.py]: filed #81

**Pull request merged**